llvm/lib/Target/AIE/AIE2InstructionSelector.cpp

```diff
@@ -188,10 +195,14 @@ class AIE2InstructionSelector : public InstructionSelector {
                                     AddressingModeInfo &AMI,
                                     MachineRegisterInfo &MRI);
   bool selectG_AIE_STORE_SRS(MachineInstr &StoreI, MachineRegisterInfo &MRI);
-  bool select512BitG_AIE_STORE_SRS(LoadStoreOpcodes &LSO,
-                                   AddressingModeInfo &AMI, Register SrcReg,
-                                   Register ShftReg, Register SignReg,
-                                   bool ConstantSign, MachineRegisterInfo &MRI);
+  bool selectG_AIE_STORE_CONCAT_SRS(MachineInstr &StoreI,
+                                    MachineRegisterInfo &MRI);
+  std::tuple<Register, Register> split512BitReg(Register SrcReg,
+                                                MachineRegisterInfo &MRI);
+  bool select512BitG_AIE_STORE_SRS(StoreSRS512BitInfo SSI,
```

Comment:
Hi @abhinay-anubola, I think you can just have two StoreSRS512BitInfo parameters here SSILo and SSIHi, in this way you can simplify your logic a a lot. You can pass twice even if they are the same.